docs/source/form-config.rst

```diff
@@ -0,0 +1,78 @@
+Form configuration files
+========================
+
+
+Introduction
+------------
+
+Form configuration files are an implementation-independent means of specifying the content of Impact Stack forms; by using JSON syntax to specify parent-child relationships between form components and key-value relationships between data, front- and back-end routines can use the same structure for their own purposes, e.g. to generate corresponding HTML or to validate form submissions.
+
+
+Structure
+---------
+
+The structure of a form configuration file is as follows:
+
+- Form components are represented as nodes in a tree structure, where the root node typically represents the component that generates a ``<form>`` element.
+- The root node **must** have a ``meta`` attribute in which to store key-value data pertaining to its use with logCRM.
```

Comment:
Original idea was "metadata" but we’ve decided on "context" now, mainly because the dict with meta information about API requests to the formtool backend and wagtail CMS is already called "meta". I don’t think it needs to be required.